packages/amplify-graphql-api-construct/src/internal/cdk-shim.ts

```diff
@@ -0,0 +1,145 @@
+  const {
+    schema,
+    resolvers,
+    functions,
+    rootStack,
+    stacks,
+  } = params;
+
+  const resolverDir = fs.mkdtempSync(path.join(assetDir, RESOLVERS_DIRECTORY));
+
+  const schemaPath = path.join(assetDir, SCHEMA_FILE_NAME);
+  fs.writeFileSync(schemaPath, schema);
+  const schemaAsset = new Asset(scope, 'schema-asset', { path: schemaPath });
+  const schemaResourceKey = Object.entries(rootStack.Resources ?? {}).find(([_, resource]) => resource.Type === 'AWS::AppSync::GraphQLSchema')?.[0];
+  const definitionLocation = ['s3:/', schemaAsset.s3BucketName, schemaAsset.s3ObjectKey].join('/');
+  rootStack.Resources![schemaResourceKey!].Properties.DefinitionS3Location = definitionLocation;
+
+  // There's probably a more elegant way to do this, but basically rewrite any sort of paths
+  // (be they asset or otherwise) in order to point them to the asset directory for the CDK.
+  Object.entries(resolvers).forEach(([resolverName, resolverCode]) => {
+    const resolverPath = path.join(resolverDir, resolverName);
+    fs.writeFileSync(resolverPath, resolverCode);
+    const currSchemaAsset = new Asset(scope, `resource-${resolverName}-asset`, { path: resolverPath });
+    const currDefinitionLocation = ['s3:/', currSchemaAsset.s3BucketName, currSchemaAsset.s3ObjectKey].join('/');
+    Object.keys(stacks).forEach((stackName) => {
+      /* eslint-disable max-len */
+      const requestResponseMappingResourceKey = Object.entries(stacks[stackName].Resources ?? {}).find(([_, resource]) => resource?.Properties?.RequestMappingTemplateS3Location && JSON.stringify(resource.Properties.RequestMappingTemplateS3Location).includes(resolverName))?.[0];
+      const responseResponseMappingResourceKey = Object.entries(stacks[stackName].Resources ?? {}).find(([_, resource]) => resource?.Properties?.ResponseMappingTemplateS3Location && JSON.stringify(resource.Properties.ResponseMappingTemplateS3Location).includes(resolverName))?.[0];
+      const codeResourceKey = Object.entries(stacks[stackName].Resources ?? {}).find(([_, resource]) => resource?.Properties?.CodeS3Location && JSON.stringify(resource.Properties.CodeS3Location).includes(resolverName))?.[0];
+      /* eslint-enable max-len */
+
+      if (requestResponseMappingResourceKey) {
+        stacks[stackName].Resources![requestResponseMappingResourceKey!].Properties
+          .RequestMappingTemplateS3Location = currDefinitionLocation;
+      }
+      if (responseResponseMappingResourceKey) {
+        stacks[stackName].Resources![responseResponseMappingResourceKey!].Properties
+          .ResponseMappingTemplateS3Location = currDefinitionLocation;
+      }
+      if (codeResourceKey) {
+        stacks[stackName].Resources![codeResourceKey!].Properties
+          .CodeS3Location = currDefinitionLocation;
+      }
+    });
+  });
+
+  // Look for AWS::Lambda::Function, and rewrite the path.
+  // This is just terrible.
+  Object.entries(functions).forEach(([functionName, functionPath]) => {
```

Comment:
is this only rewriting lambdas that are created internally to this construct? What would be the LOE to not add "this is just terrible" code?

Reply:
This is rewriting any file-based asset. LOE is pretty high, since it's going to require us to refactor a decent portion of the transformer core to no longer internally synth, and rewrite the internal StackManagement, RootStack, Nestedstack, and StackSynthesizers, but it's on the roadmap after we get the initial draft out for review.